### PR TITLE
[4.4] Container Database

### PR DESCRIPTION
**Pull request by heelc29**
### Summary of Changes

Use ClassName instead of hardcoded string

Joomla\Database\DatabaseInterface is also used. What should be used: `DatabaseDriver` or `DatabaseInterface `

### Testing Instructions

Use your website (for eg. download the 'System Information' or use Webauthn)

Code Review

### CODE REVIEW

**Comment by laoneo** [comment]
Can you change them to the DatabaseInterface::class? Better to use the interface, so we can put into the container anything which implements it and do not add the requirement to be a driver.

**Review by wilsonge — APPROVED** [review]

**Comment by heelc29** [comment]
> Can you change them to the DatabaseInterface::class?

@laoneo Done

File: administrator/components/com_admin/src/Model/SysinfoModel.php [hunk 1]
```diff
@@ -449,7 +450,7 @@ public function getPhpInfoArray(): array
 	public function getExtensions(): array
 	{
 		$installed = [];
-		$db = Factory::getContainer()->get('DatabaseDriver');
+		$db = Factory::getContainer()->get(DatabaseInterface::class);
```
Comment (laoneo) [inline administrator/components/com_admin/src/Model/SysinfoModel.php]:
```suggestion
		$db = $this->getDatabase();
```

Comment (heelc29) [reply administrator/components/com_admin/src/Model/SysinfoModel.php]: This PR is for 4.1, here is only `getDbo` available

**Comment by laoneo** [comment]
Can you rebase this pr to 4.2 please? There we should use the injected database whenever possible.

**Comment by heelc29** [comment]
> Can you rebase this pr to 4.2 please?

@laoneo Done.

**Review by laoneo — APPROVED** [review]

**Comment by joomdonation** [comment]
I have tested this item :white_check_mark: successfully on c15f5a85066160c2ea18cb3820f444e596172a1a<br /><br />Applied patch, access to various functions like System Information, Users Manage, Articles Manage, Add/edit user/article, view article where tags displayed...., nothing is broken. Done a code review, too.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37844">issues.joomla.org/tracker/joomla-cms/37844</a>.</sub>

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 4.3-dev.

**Comment by laoneo** [comment]
Can you rebase to 4.4? Thanks for your patience...

**Comment by laoneo** [comment]
Now there is a conflict. Once solved, I will merge it.

**Comment by heelc29** [comment]
> Fixed it.

Thanks

**Comment by laoneo** [comment]
Thanks!

**Pull request merged**